byron/ledger/executable-spec/src/Ledger/Update.hs

```diff
@@ -718,15 +716,15 @@ delegationMap (_, dms, _, _) = dms
 -- | The update interface state is shared amongst various rules, so we define it
 -- as an alias here.
 type UPIState =
-  ( (ProtVer, PParams)
-  , [(Core.Slot, (ProtVer, PParams))]
-  , Map ApName (ApVer, Core.Slot, Metadata)
-  , Map UpId (ProtVer, PParams)
-  , Map UpId (ApName, ApVer, Metadata)
-  , Map UpId Core.Slot
-  , Set (UpId, Core.VKeyGenesis)
-  , Set (ProtVer, Core.VKeyGenesis)
-  , Map UpId Core.Slot
+  ( (ProtVer, PParams) -- (pv, pps)
```

Comment:
I smell a record ;)

Reply:
But this was the 9-tuple I had to write Goblin instances for! We can't let all that somewhat-generic code go to waste! 😝

Reply:
Hehe, yeah, records are coming :snowflake: